### ISSUE TITLE
apropos(perm) gives [...,permult,...] and 'permult'  is non-existent command or symbol

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#3038]
Imported from SourceForge on 2024-07-07 09:35:59
Created by **[zmth](https://sourceforge.net/u/zmth/)** on 2018-10-02 16:19:43
Original: https://sourceforge.net/p/maxima/bugs/3468

---

wxMaxima version: 16.4.2
Maxima version: 5.38.1_5_gdf93b7b_dirty
Maxima build date: 2016-05-13 22:33:49
Host type: i686-w64-mingw32
System type: 
gcc -mno-cygwin -g -O2 -W -Wswitch -Wcomment -Wpointer-arith -Wimplicit -Wreturn-type -Wmissing-declarations -Wno-sign-compare -Wno-format-nonliteral -O2 -fexpensive-optimizations -falign-functions=4 -D_WIN32 -DENABLE_UNICODE -I/usr/local/include -DDYNAMIC_FFI -I. -L/usr/local/lib  -lintl /usr/local/lib/libreadline.dll.a -L/usr/local/lib -ltermcap /usr/local/lib/libavcall.a /usr/local/lib/libcallback.a -luser32 -lws2_32 -lole32 -loleaut32 -luuid -liconv -L/usr/local/lib -lsigsegv libgnu_cl.a 
SAFETY=0 HEAPCODES STANDARD_HEAPCODES GENERATIONAL_GC SPVW_BLOCKS SPVW_MIXED TRIVIALMAP_MEMORY
libsigsegv 2.8
libiconv 1.13
libreadline 6.0 GNU C 3.4.5 (mingw-vista special r3) PC/386
Lisp implementation type: CLISP
Lisp implementation version: 2.49 (2010-07-07) (built on toshiba [192.168.43.3])

In wxmaxima:

apropos("perm")
gives [..., permult,....]

That is it gives permult as one of the members in the list as if that is a command or symbol or something in wxmaxima which it is NOT. NOwhere is ' permult ' in the index or help file or anything.
I know in the commercial Macsyma ' permult ' with arguments IS a command for multiplying two permutations which is handy - eg permult([2,1],[2,1]) gives [1,2]. Too bad it is not a command in
wxmaxima. Also why is the help in the later version of maxima NOT as it is in version 5.28 where all have to do is alt-h and it comes up but not
in version 5.38. Unfortunately one has to use the mouse. Also if in index of help in version 5.28 just type the first or first few letters and the whole list
of that starting is given all down the screen automatically for one to choose but NOT in version 5.38 - very inconvenient. Why is HELP made 
WORSE in the later version of wxmaxima ?

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 09:36:01
Created by **[peterpall](https://sourceforge.net/u/peterpall/)** on 2018-10-02 16:45:20
Original: https://sourceforge.net/p/maxima/bugs/3468/#a481

---

You are using more than 2 years old versions of maxima+wxMaxima. If you upgrade to the current one the command name is shown as "permut" without "l" and the help should work.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 09:36:04
Created by **[peterpall](https://sourceforge.net/u/peterpall/)** on 2018-10-02 16:50:16
Original: https://sourceforge.net/p/maxima/bugs/3468/#ba82

---

Or if you meant permult from the combinatorics package:
~~~
?? perm

 0: perm_cycles  (Functions and Variables for Combinatorics)
 1: perm_decomp  (Functions and Variables for Combinatorics)
 2: perm_inverse  (Functions and Variables for Combinatorics)
 3: perm_length  (Functions and Variables for Combinatorics)
 4: perm_lex_next  (Functions and Variables for Combinatorics)
 5: perm_lex_rank  (Functions and Variables for Combinatorics)
 6: perm_lex_unrank  (Functions and Variables for Combinatorics)
 7: perm_next  (Functions and Variables for Combinatorics)
 8: perm_parity  (Functions and Variables for Combinatorics)
 9: perm_rank  (Functions and Variables for Combinatorics)
 10: perm_undecomp  (Functions and Variables for Combinatorics)
 11: perm_unrank  (Functions and Variables for Combinatorics)
 12: permanent  (Functions and Variables for Matrices and Linear Algebra)
 13: permp  (Functions and Variables for Combinatorics)
 14: perms  (Functions and Variables for Combinatorics)
 15: perms_lex  (Functions and Variables for Combinatorics)
 16: permult  (Functions and Variables for Combinatorics)
 17: permut  (Functions and Variables for Symmetries)
 18: permutation  (Package functs)
 19: permutations  (Functions and Variables for Sets)
 20: permute  (Functions and Variables for Combinatorics)
 21: random_perm  (Functions and Variables for Combinatorics)
 22: random_permutation  (Functions and Variables for Sets)
~~~
Actually I've spent much time for improving the manual in the last 2 years. Not that I've received much thanks for that.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 09:36:08
Created by **[peterpall](https://sourceforge.net/u/peterpall/)** on 2018-10-02 16:53:23
Original: https://sourceforge.net/p/maxima/bugs/3468/#6f2b

---

Apropos might not find permult since permult isn't one of the few first commands that are most similar to "perm": For example it is much shorter. 

My guess is that commands with more similar names were introduced recently so the apropos() command returns their names now.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 09:36:11
Created by **[zmth](https://sourceforge.net/u/zmth/)** on 2018-10-11 22:56:34
Original: https://sourceforge.net/p/maxima/bugs/3468/#6110

---

I know ' permut ' was also in the list along with permult and many others.
So how do i get to use that combinatorics package. Yea i could write fct's to do the multiply permutations but it is handy to already have that.  NOte also that ' full screen ' i tried and now there was no way to get it back to before and nowhere is  the topic ' full screen ' even in the manual atleast now my manuals - does not come up as a topic

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 09:36:15
Created by **[zmth](https://sourceforge.net/u/zmth/)** on 2018-10-11 23:16:49
Original: https://sourceforge.net/p/maxima/bugs/3468/#d1a9

---

also seems cannot copy from command line and paste into whatever editor one is using to write... and work with macsyma as u play hell using the one in maxima - not even lined up correct.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 09:36:18
Created by **[peterpall](https://sourceforge.net/u/peterpall/)** on 2018-10-12 05:19:42
Original: https://sourceforge.net/p/maxima/bugs/3468/#951a

---

That would mean that you currently are using an old wxMaxima. Please update to the newest version; If it is a wxMaxima bug feel free to file a bug at https://github.com/wxMaxima-developers/wxmaxima

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 09:36:22
Created by **[peterpall](https://sourceforge.net/u/peterpall/)** on 2018-10-12 05:21:47
Original: https://sourceforge.net/p/maxima/bugs/3468/#6110/934d

---

"Full screen" mode is provided by the operating system and I hope it does provide a way to leave it again. My guess would be it can be left using Alt+Tab or Alt+Enter.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 09:36:26
Created by **[peterpall](https://sourceforge.net/u/peterpall/)** on 2018-10-12 05:23:11
Original: https://sourceforge.net/p/maxima/bugs/3468/#6110/7811

---

What you need is a working manual => please try to use the following one: maxima.sourceforge.net/docs/manual/maxima_singlepage.html

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 09:36:29
Created by **[villate](https://sourceforge.net/u/villate/)** on 2018-10-12 07:32:14
Original: https://sourceforge.net/p/maxima/bugs/3468/#7230

---

The combinatorics package is an optional package that I created a few months ago. As such, it has only become available in the latest release (5.42.0) and you have to use "load" first, before using permult, as the manual tells you:
~~~
(%i1) ? permult


 -- Function: permult (<p_1>, ..., <p_m>)

     Returns the product of two or more permutations <p_1>, ..., <p_m>.

     Example:

          (%i1) load("combinatorics")$
          (%i2) permult ([2,3,1], [3,1,2], [2,1,3]);
          (%o2)                        [2, 1, 3]
~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 09:36:33
Created by **[zmth](https://sourceforge.net/u/zmth/)** on 2019-05-12 16:09:19
Original: https://sourceforge.net/p/maxima/bugs/3468/#a481/4d07

---

do i have to update both maxima and wxmaxima ? or just one or the other or is one included in the other ? It's a bit unclear as i get this message somewhere 
" The latest version of wxMaxima is 19.05.3. All files released so far are available from Github. "
